README.md

```diff
@@ -86,69 +98,61 @@ const client = sanityClient({ /* ... */});
 export const runQuery = makeSafeQueryRunner(query => client.fetch(query));
 
 // Now you can fetch your query's result, and validate the response, all in one.
-const data = await runQuery(q("*", q.filter("_type == 'pokemon'"), /* ... */));
+const data = await runQuery(q("*").filter("_type == 'pokemon'"));
 ```
 
-### `q.grab`
+### `.grab`
 
-Handles [projections](https://www.sanity.io/docs/how-queries-work#727ecb6f5e15), or selecting fields from an existing set of documents. This is one of the primary mechanisms for providing a schema for the data you expect to get. 
+Available on `UnknownResult`, `ArrayResult`, and `EntityResult`, handles [projections](https://www.sanity.io/docs/how-queries-work#727ecb6f5e15), or selecting fields from an existing set of documents. This is the primary mechanism for providing a schema for the data you expect to get. 
 
 `q.grab` accepts a "selection" object as its sole argument, with three different forms:
 
 ```ts
-q(
-  "*",
-  q.grab({
-    // projection is `{ "name": name }`, and validates that `name` is a string.
-    name: ['name', q.string()],
-    
-    // shorthand for `description: ['description', q.string()]`,
-    //  projection is just `{ description }`
-    description: q.string(),
-    
-    // can also pass a sub-query for the field,
-    //  projection is `{ "types": types[]->{ name } }`
-    types: q("types", q.filter(), q.deref(), q.grab({ name: q.string() }))
-  }),
-);
+q("*").grab({
+  // projection is `{ "name": name }`, and validates that `name` is a string.
+  name: ['name', q.string()],
+
+  // shorthand for `description: ['description', q.string()]`,
+  //  projection is just `{ description }`
```

Comment:
groq offers a shorthand where instead of writing `{ "name": name }` you can write `{ name }`. In my experience, I see most people prefer the shorthand – unless they need to rename a field or do something more complex than just grab a field.

I'm not sure there'd be much benefit in expanding `{ description }` out to `{ "description": description }`, it's mostly just added bytes. Maybe the example `name: ['name', q.string()],` is just a bit too simplified. The reason that first form is an option is so you can do something like:

```ts
q("*").grab({ name: ["coalesce(name, 'no name')", q.string()] });
```

or something of the likes.

Reply:
I see. So it's not so much shorthand within this library (which produces different GROQ for each syntax) as it is shorthand upon being used in GROQ.

Reply:
Ummm I think it might be _both_ – a shorthand used within this library (because `name: q.string()` is nicer to write than `name: ["name", q.string()]`) that also maps nicely into an official groq shorthand (using `{ name }` instead of `{ "name": name }` for the projection).

Reply:
Makes sense 👍